### PR TITLE
ADDING MORE TEST FOR std::string ON ALL PLATFORMS

### PR DESCRIPTION
**Pull request by Emmankoko**

### CODE REVIEW

File: source/stdcpp/test/string.d [hunk 1]
```diff
@@ -23,5 +23,20 @@ unittest
...
-    assert(a.capacity == stringCapacity("hello"));
+    assert(a.capacity == stringCapacity("helloa"));
+    assert(a.front() == 'h');
+    assert(a.back() == 'a');
+    a.resize(4); // shrinks a to "hell"
+    assert(a.size() == 4);
+    auto b = std_string("Hi, this is a test for string capacity growth for a length more than the base SSO");
+    assert(b.capacity == stringCapacity("Hi, this is a test for string capacity growth for a length more than the base SSO"));
+    a.swap(b); // a and b swaps
+    assert(a.capacity == stringCapacity("Hi, this is a test for string capacity growth for a length more than the base SSO"));
```
Comment (Geod24) [inline source/stdcpp/test/string.d]:
```suggestion
    immutable LongStr = "Hi, this is a test for string capacity growth for a length more than the base SSO";
    auto b = std_string(LongStr);
    assert(b.capacity == stringCapacity(LongStr.ptr));
    a.swap(b); // a and b swaps
    assert(a.capacity == stringCapacity(LongStr.ptr));
```

File: source/stdcpp/test/string.d [hunk 2]
```diff
@@ -23,5 +23,20 @@ unittest
...
+    assert(b.capacity == stringCapacity("hell")); // a was shrinked to hell so b contains 'hell'
+    b.pop_back();
+    assert(b.size() == 3);
+    assert(a.empty == 0);
+    a.clear();
+    assert(a.empty == 1);
+
```
Comment (Geod24) [inline source/stdcpp/test/string.d]:
```suggestion
```

Comment (Emmankoko) [reply source/stdcpp/test/string.d]: why? Unecessary?

Comment (Geod24) [reply source/stdcpp/test/string.d]: Yes, why would you have trailing whitespace ?

**Pull request merged**